app/views/ems_infra/_form.html.haml

```diff
@@ -72,51 +62,7 @@
-        = _('Host Default VNC Start Port')
-      .col-md-8
-        %input.form-control{"type"        => "text",
-                            "id"          => "ems_vnc_start_port",
-                            "name"        => "host_default_vnc_port_start",
-                            "ng-model"    => "emsCommonModel.host_default_vnc_port_start",
-                            "maxlength"   => 6,
-                            "checkchange" => ""}
-
-    .form-group{"ng-class" => "{'has-error': angularForm.host_default_vnc_port_end.$invalid}", "ng-if" => "emsCommonModel.emstype == 'vmwarews'"}
-      %label.col-md-2.control-label{"for" => "ems_vnc_end_port"}
-        = _('Host Default VNC End Port')
-      .col-md-8
-        %input.form-control{"type"        => "text",
-                            "id"          => "ems_vnc_end_port",
-                            "name"        => "host_default_vnc_port_end",
-                            "ng-model"    => "emsCommonModel.host_default_vnc_port_end",
-                            "maxlength"   => 6,
-                            "checkchange" => ""}
-
-  %hr
-  - if controller_name == "ems_container"
-    = render :partial => "/layouts/container_auth", :locals  => {:record => @ems}
-  - else
-    %div{"ng-show" => "emsCommonModel.emstype != '' && emsCommonModel.emstype == 'azure'"}
-      %h3
-        = _("Credentials")
-      = render :partial => "/layouts/angular-bootstrap/auth_credentials_angular_bootstrap",
-               :locals  => {:record                 => @ems,
-                            :ng_model               => "emsCommonModel",
-                            :ng_reqd_password       => "emsCommonModel.default_userid != ''",
-                            :prefix                 => "default",
-                            :validate_url           => @ems.id ? "update" : "create",
-                            :id                     => @ems.id || "new",
-                            :basic_info_needed      => true,
-                            :userid_label           => _("Client ID"),
-                            :password_label         => _("Client Key"),
-                            :verify_label           => _("Confirm Client Key"),
-                            :passwd_mismatch        => _("Client Keys do not match"),
-                            :change_stored_password => _("Change stored client key"),
-                            :cancel_password_change => _("Cancel client key change"),
-                            :verify_title_off       => _("Tenant ID, Client ID and matching Client Key fields are needed to perform verification of credentials")}
-    %div{"ng-show" => "emsCommonModel.emstype != 'azure'"}
-      = render :partial => "/layouts/angular/multi_auth_credentials", :locals  => {:record => @ems, :ng_model => "emsCommonModel"}
+  %div{"ng-if"      => "!!emsCommonModel.emstype",
+       "ng-include" => "'/ems_infra/properties?id=#{@ems.id}&type=' + emsCommonModel.emstype"}
```

Comment:
imho you should use a rails route helper instead (in case the route change in the future)

Reply:
btw: if you pass the id, do you still need the type?

Reply:
Being strict the `type` isn't needed if the `id` is provided, as it can be inferred from the `id` (if there is really an object corresponding to that `id`). However these _ifs_ are details that the client doesn't need to know.

Reply:
@ohadlevy Not sure what you mean by "using a rails route helper", can you give me an example?

Reply:
@ohadlevy I am not sure if that can be used as the `properties` isn't a resource. However, if the concern is that the path can change, I can add something like this to the controller:

```ruby
included do
   ...
   helper_method :properties_path
end
```

And then:

```ruby
private

def properties_path
  '/ems_infra/properties'
end
```

Finally, in the `.haml` I can do this:

```haml
%div{"ng-if"      => ...,
     "ng-include" => "'#{properties_path}?..."}
```

Would that address your concern?